File: Models/PlacementProgress.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;
using MtpApp.Models;
using System.ComponentModel.DataAnnotations.Schema;

namespace MtpApp.Models
{
    public class PlacementProgress
    {
        public int Id { get; set; }
        [Required]
        public string Completed { get; set; }
        [ForeignKey("Monitoring")]
        public int MonitoringId { get; set; }
        public Monitoring Monitoring { get; set; }
        [Required]
        public string PlacementStatus { get; set; }
        [Required]
        public string Salary { get; set; }
        public string Note { get; set; }     
    }
}

File: spring-backend/src/main/java/com/mtp/api/models/PlacementProgress.java
package com.mtp.api.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "PlacementProgresses")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PlacementProgress {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    private String completed;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "MonitoringId")
    private Monitoring monitoring;

    @NotNull
    private String placementStatus;

    @NotNull
    private String salary;

    private String note;
}
